features/poller/shell/src/main/java/org/opennms/netmgt/poller/shell/Poll.java

```diff
@@ -44,60 +47,147 @@
+import org.opennms.netmgt.dao.api.NodeDao;
+import org.opennms.netmgt.model.OnmsIpInterface;
+import org.opennms.netmgt.model.OnmsNode;
 import org.opennms.netmgt.poller.LocationAwarePollerClient;
+import org.opennms.netmgt.poller.MonitoredService;
 import org.opennms.netmgt.poller.PollStatus;
 import org.opennms.netmgt.poller.PollerResponse;
+import org.opennms.netmgt.poller.ServiceMonitor;
 import org.opennms.netmgt.poller.support.SimpleMonitoredService;
+import org.springframework.transaction.TransactionStatus;
+import org.springframework.transaction.support.TransactionCallback;
+import org.springframework.transaction.support.TransactionOperations;
 
-@Command(scope = "poller", name = "poll", description = "Used to invoke a monitor against a host at a specified location")
+@Command(scope = "opennms-poller", name = "poll", description = "Used to invoke a monitor against a host at a specified location")
 @Service
 public class Poll implements Action {
 
     @Option(name = "-l", aliases = "--location", description = "Location", required = false, multiValued = false)
-    String location;
+    String location = "Default";
+
+    @Option(name = "-S", aliases = "--service", description = "Service name", required = false, multiValued = false)
+    String serviceName;
+
+    @Option(name = "-P", aliases = "--package", description = "Poller Package", required = false, multiValued = false)
+    String packageName;
 
     @Option(name = "-s", aliases = "--system-id", description = "System ID")
     String systemId;
 
     @Option(name = "-t", aliases = "--ttl", description = "Time to live", required = false, multiValued = false)
     Long ttlInMs;
-    
+
     @Option(name = "-n", aliases = "--node-id", description = "Node Id for Service", required = false, multiValued = false)
     int nodeId;
 
-    @Argument(index = 0, name = "monitorClass", description = "Monitor class", required = true, multiValued = false)
+    @Option(name = "-c", aliases = "--class", description = "Monitor Class", required = false, multiValued = false)
     @Completion(MonitorClassNameCompleter.class)
     String className;
 
-    @Argument(index = 1, name = "host", description = "Hostname or IP Address of the system to poll", required = true, multiValued = false)
+    @Argument(index = 0, name = "host", description = "Hostname or IP Address of the system to poll", required = true, multiValued = false)
     String host;
 
-    @Argument(index = 2, name = "attributes", description = "Monitor specific attributes in key=value form", multiValued = true)
+    @Argument(index = 1, name = "attributes", description = "Monitor specific attributes in key=value form", multiValued = true)
     List<String> attributes;
 
     @Reference
     public LocationAwarePollerClient locationAwarePollerClient;
 
+    @Reference
+    public NodeDao nodeDao;
+
+    @Reference
+    public IpInterfaceDao ipInterfaceDao;
+
+    @Reference
+    public TransactionOperations transactionTemplate;
```

Comment:
Can you use `SessionUtils` instead? 
Using `TransactionOperations` may result in weird bundle states when `bundle:watch *` is enabled or a bundle is trying to refresh.